protocol/storage/miner.go

```diff
@@ -375,23 +379,29 @@ func rejectProposal(sm *Miner, p *storagedeal.Proposal, reason string) (*storage
 }
 
 func (sm *Miner) updateDealResponse(ctx context.Context, proposalCid cid.Cid, f func(*storagedeal.Response)) error {
```

Comment:
There needs to be a comment explaining what the purpose of `f` is.  The way it's called in the code does not help.